### ISSUE TITLE
Semantics focus moves to first element after the focused widget is updated

### ISSUE THREAD

**Issue by Stewioie** [flutter/flutter#80271]
### Example code

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _value;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
        leading: IconButton(icon: Icon(Icons.arrow_back), onPressed: () {}),
      ),
      body: Center(
        child: Padding(
            padding: EdgeInsets.symmetric(horizontal: 20),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: <Widget>[
                DropdownButtonFormField(
                    value: _value,
                    onChanged: (value) {
                      setState(() {
                        _value = value;
                      });
                    },
                    hint: Text("Open dropdown to select an item"),
                    items: [
                      DropdownMenuItem(child: Text("Dummy 1"), value: 1),
                      DropdownMenuItem(child: Text("Dummy 2"), value: 2),
                      DropdownMenuItem(child: Text("Dummy 3"), value: 3)
                    ])
              ],
            )),
      ),
    );
  }
}

```

### Steps

- Turn on accessibility (Talkback) in Settings
- Start App
- Use accessibility navigation to select an item from Dropdown
- The accessibility focus jumps to first item on page (Back button)

### Expected results

The accessibility focus stays on focused widget

### Flutter doctor output

[✓] Flutter (Channel stable, 2.0.3, on macOS 11.1 20C69 darwin-x64, locale hu-HU)
    • Flutter version 2.0.3 at /Users/developer/Work/flutter
    • Framework revision 4d7946a68d (4 weeks ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/developer/Work/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/developer/Work/sdk
    • ANDROID_SDK_ROOT = /Users/developer/Work/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • SM G965F (mobile) • 279d5774421c7ece • android-arm64  • Android 10 (API 29)
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 89.0.4389.114

• No issues found!

**Comment by darshankawar** [comment]
Issue replicable using latest stable. After selecting an item from dropdown, the item is selected and displayed in the dropdown, but focus moves to back button by default.

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.0.4, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.0.4 at /Users/dhs/documents/fluttersdk/flutter
    • Framework revision b1395592de (7 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)    

[✓] VS Code (version 1.55.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (3 available)
	  • SM A260G (mobile)            • 5200763ebcfa861f • android-arm    • Android
      8.1.0 (API 27)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 14.4.1
    • macOS (desktop)           • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 89.0.4389.114

! Doctor found issues in 1 category.

```

</details>

**Comment by exaby73** [comment]
This issue is still pre-producible on the latest stable and master versions. Though there is a small difference in the issue now. Instead of the back button getting focused, the title in the `AppBar` is focused. See the attached video to see this behavior.

<details><summary>Code sample used</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int? _value;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
        leading: IconButton(
          onPressed: () {},
          icon: const Icon(Icons.arrow_back),
        ),
      ),
      body: Center(
        child: Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20),
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              DropdownButtonFormField<int>(
                value: _value,
                onChanged: (value) {
                  setState(() {
                    _value = value;
                  });
                },
                hint: const Text("Open dropdown to select an item"),
                items: const [
                  DropdownMenuItem(child: Text("Dummy 1"), value: 1),
                  DropdownMenuItem(child: Text("Dummy 2"), value: 2),
                  DropdownMenuItem(child: Text("Dummy 3"), value: 3),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

</details>

<details><summary>Video</summary>

https://user-images.githubusercontent.com/50555895/167816629-d8671862-c74f-4a10-9dc2-d772cbb778a3.mov

</details>

<details><summary><code>flutter doctor -v</code></summary>

```
[✓] Flutter (Channel master, 2.13.0-0.0.pre.1026, on macOS 12.3.1 21E258 darwin-arm, locale en-US)
    • Flutter version 2.13.0-0.0.pre.1026 at /Users/nabeelparkar/fvm/versions/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c1b909e3fc (4 hours ago), 2022-05-10 22:49:08 -0700
    • Engine revision 70c5bdfa3b
    • Dart version 2.18.0 (build 2.18.0-103.0.dev)
    • DevTools version 2.13.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0-rc1)
    • Android SDK at /Users/nabeelparkar/Library/Android/sdk/
    • Platform android-32, build-tools 32.0.0-rc1
    • ANDROID_SDK_ROOT = /Users/nabeelparkar/Library/Android/sdk/
    • Java binary at: /opt/homebrew/opt/openjdk@11/bin/java
    • Java version OpenJDK Runtime Environment Homebrew (build 11.0.15+0)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] IntelliJ IDEA Ultimate Edition (version 2022.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 67.0.4
    • Dart plugin version 221.5588

[✓] VS Code (version 1.67.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (3 available)
    • V2050 (mobile)  • 3085425058000NM • android-arm64  • Android 12 (API 31)
    • macOS (desktop) • macos           • darwin-arm64   • macOS 12.3.1 21E258 darwin-arm
    • Chrome (web)    • chrome          • web-javascript • Brave Browser 101.1.38.111

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

**Comment by huycozy** [comment]
Re-verifying this on the latest stable `3.35.2` and master `3.36.0-1.0.pre-286` with sample code at https://github.com/flutter/flutter/issues/80271#issuecomment-1123421571, I see behavior has been changed:

After tapping on the 2nd item, the reader reads the title on AppBar first, then reads the 2nd item; the focus (highlighted/green rect) persists at the Dropdown, but it is blinking twice.

I also tested with [Menu](https://developer.android.com/develop/ui/compose/components/menu) in Jetpack Compose to compare both: after tapping on other item, the reader does not read the title on appbar but reads app name (or window name?); it does not show blinking effect as in Flutter, though.

By the behavior changes and comparison, I think the current result matches the expectation. @Stewioie Can you please confirm on your end as well? 

If we need to solve the remaining issues (blinking and reading order), I would suggest filing separate issues for better tracking. In such cases, it's better to have accessibility team's suggestion to understand the proper expectations. 

<details open>
<summary>Demo</summary>

| Flutter | Android Jetpack Compose |
| --------------- | --------------- |
<video src="https://github.com/user-attachments/assets/a5a75e2b-5968-48a8-8595-7f9d2f3a41f6"/> | <video src="https://github.com/user-attachments/assets/b5ad2997-e24b-42f1-8960-e885ec7400e8"/>

</details>

**Comment by huycozy** [comment]
@Stewioie I'm going ahead and closing this issue. I also filed https://github.com/flutter/flutter/issues/174698 for the blinking issue. Please comment or file another issue if you still encounter this issue.
